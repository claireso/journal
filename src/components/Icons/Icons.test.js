// import React from 'react'
import { render } from '@testing-library/react'

import * as Icons from './index'

describe('<Icons />', () => {
  test('should render <IconPlus />', () => {
    const { container } = render(<Icons.IconPlus />)

    expect(container).toMatchSnapshot()
  })

  test('should render <IconPlus /> with custom sizes', () => {
    const { container } = render(<Icons.IconPlus width="20" height="20" />)

    expect(container).toMatchSnapshot()
  })

  test('should render <IconBack />', () => {
    const { container } = render(<Icons.IconBack />)

    expect(container).toMatchSnapshot()
  })

  test('should render <IconBack /> with custom sizes', () => {
    const { container } = render(<Icons.IconBack width="20" height="20" />)

    expect(container).toMatchSnapshot()
  })

  test('should render <IconUpload />', () => {
    const { container } = render(<Icons.IconUpload />)

    expect(container).toMatchSnapshot()
  })

  test('should render <IconUpload /> with custom sizes', () => {
    const { container } = render(<Icons.IconUpload width="20" height="20" />)

    expect(container).toMatchSnapshot()
  })

  test('should render <IconClose />', () => {
    const { container } = render(<Icons.IconClose />)

    expect(container).toMatchSnapshot()
  })

  test('should render <IconClose /> with custom sizes', () => {
    const { container } = render(<Icons.IconClose width="20" height="20" />)

    expect(container).toMatchSnapshot()
  })

  test('should render <IconAngleRight />', () => {
    const { container } = render(<Icons.IconAngleRight />)

    expect(container).toMatchSnapshot()
  })

  test('should render <IconAngleRight /> with custom sizes', () => {
    const { container } = render(<Icons.IconAngleRight width="20" height="20" />)

    expect(container).toMatchSnapshot()
  })

  test('should render <IconPencil />', () => {
    const { container } = render(<Icons.IconPencil />)

    expect(container).toMatchSnapshot()
  })

  test('should render <IconPencil /> with custom sizes', () => {
    const { container } = render(<Icons.IconPencil width="20" height="20" />)

    expect(container).toMatchSnapshot()
  })

  test('should render <IconDelete />', () => {
    const { container } = render(<Icons.IconDelete />)

    expect(container).toMatchSnapshot()
  })

  test('should render <IconDelete /> with custom sizes', () => {
    const { container } = render(<Icons.IconDelete width="20" height="20" />)

    expect(container).toMatchSnapshot()
  })
})
