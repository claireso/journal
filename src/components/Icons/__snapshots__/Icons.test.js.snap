// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icons /> should render <IconAngleRight /> 1`] = `
<div>
  <svg
    height="14"
    viewBox="0 0 9 28"
    width="9"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M9.297 15c0 0.125-0.063 0.266-0.156 0.359l-7.281 7.281c-0.094 0.094-0.234 0.156-0.359 0.156s-0.266-0.063-0.359-0.156l-0.781-0.781c-0.094-0.094-0.156-0.219-0.156-0.359 0-0.125 0.063-0.266 0.156-0.359l6.141-6.141-6.141-6.141c-0.094-0.094-0.156-0.234-0.156-0.359s0.063-0.266 0.156-0.359l0.781-0.781c0.094-0.094 0.234-0.156 0.359-0.156s0.266 0.063 0.359 0.156l7.281 7.281c0.094 0.094 0.156 0.234 0.156 0.359z"
    />
  </svg>
</div>
`;

exports[`<Icons /> should render <IconAngleRight /> with custom sizes 1`] = `
<div>
  <svg
    height="20"
    viewBox="0 0 9 28"
    width="20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M9.297 15c0 0.125-0.063 0.266-0.156 0.359l-7.281 7.281c-0.094 0.094-0.234 0.156-0.359 0.156s-0.266-0.063-0.359-0.156l-0.781-0.781c-0.094-0.094-0.156-0.219-0.156-0.359 0-0.125 0.063-0.266 0.156-0.359l6.141-6.141-6.141-6.141c-0.094-0.094-0.156-0.234-0.156-0.359s0.063-0.266 0.156-0.359l0.781-0.781c0.094-0.094 0.234-0.156 0.359-0.156s0.266 0.063 0.359 0.156l7.281 7.281c0.094 0.094 0.156 0.234 0.156 0.359z"
    />
  </svg>
</div>
`;

exports[`<Icons /> should render <IconBack /> 1`] = `
<div>
  <svg
    height="16"
    viewBox="0 0 32 32"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M32 14.286v3.43q0 .25-.16.41t-.412.16H9.142v4q0 .376-.34.52t-.624-.09l-6.857-6.25q-.178-.18-.178-.41 0-.25.18-.43l6.856-6.32q.286-.25.625-.108.34.16.34.518v4h22.285q.25 0 .41.16t.162.412z"
    />
  </svg>
</div>
`;

exports[`<Icons /> should render <IconBack /> with custom sizes 1`] = `
<div>
  <svg
    height="20"
    viewBox="0 0 32 32"
    width="20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M32 14.286v3.43q0 .25-.16.41t-.412.16H9.142v4q0 .376-.34.52t-.624-.09l-6.857-6.25q-.178-.18-.178-.41 0-.25.18-.43l6.856-6.32q.286-.25.625-.108.34.16.34.518v4h22.285q.25 0 .41.16t.162.412z"
    />
  </svg>
</div>
`;

exports[`<Icons /> should render <IconClose /> 1`] = `
<div>
  <svg
    height="16"
    viewBox="0 0 22 28"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M20.281 20.656c0 0.391-0.156 0.781-0.438 1.062l-2.125 2.125c-0.281 0.281-0.672 0.438-1.062 0.438s-0.781-0.156-1.062-0.438l-4.594-4.594-4.594 4.594c-0.281 0.281-0.672 0.438-1.062 0.438s-0.781-0.156-1.062-0.438l-2.125-2.125c-0.281-0.281-0.438-0.672-0.438-1.062s0.156-0.781 0.438-1.062l4.594-4.594-4.594-4.594c-0.281-0.281-0.438-0.672-0.438-1.062s0.156-0.781 0.438-1.062l2.125-2.125c0.281-0.281 0.672-0.438 1.062-0.438s0.781 0.156 1.062 0.438l4.594 4.594 4.594-4.594c0.281-0.281 0.672-0.438 1.062-0.438s0.781 0.156 1.062 0.438l2.125 2.125c0.281 0.281 0.438 0.672 0.438 1.062s-0.156 0.781-0.438 1.062l-4.594 4.594 4.594 4.594c0.281 0.281 0.438 0.672 0.438 1.062z"
    />
  </svg>
</div>
`;

exports[`<Icons /> should render <IconClose /> with custom sizes 1`] = `
<div>
  <svg
    height="20"
    viewBox="0 0 22 28"
    width="20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M20.281 20.656c0 0.391-0.156 0.781-0.438 1.062l-2.125 2.125c-0.281 0.281-0.672 0.438-1.062 0.438s-0.781-0.156-1.062-0.438l-4.594-4.594-4.594 4.594c-0.281 0.281-0.672 0.438-1.062 0.438s-0.781-0.156-1.062-0.438l-2.125-2.125c-0.281-0.281-0.438-0.672-0.438-1.062s0.156-0.781 0.438-1.062l4.594-4.594-4.594-4.594c-0.281-0.281-0.438-0.672-0.438-1.062s0.156-0.781 0.438-1.062l2.125-2.125c0.281-0.281 0.672-0.438 1.062-0.438s0.781 0.156 1.062 0.438l4.594 4.594 4.594-4.594c0.281-0.281 0.672-0.438 1.062-0.438s0.781 0.156 1.062 0.438l2.125 2.125c0.281 0.281 0.438 0.672 0.438 1.062s-0.156 0.781-0.438 1.062l-4.594 4.594 4.594 4.594c0.281 0.281 0.438 0.672 0.438 1.062z"
    />
  </svg>
</div>
`;

exports[`<Icons /> should render <IconDelete /> 1`] = `
<div>
  <svg
    height="16"
    version="1.1"
    viewBox="0 0 22 28"
    width="22"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M8 21.5v-11c0-0.281-0.219-0.5-0.5-0.5h-1c-0.281 0-0.5 0.219-0.5 0.5v11c0 0.281 0.219 0.5 0.5 0.5h1c0.281 0 0.5-0.219 0.5-0.5zM12 21.5v-11c0-0.281-0.219-0.5-0.5-0.5h-1c-0.281 0-0.5 0.219-0.5 0.5v11c0 0.281 0.219 0.5 0.5 0.5h1c0.281 0 0.5-0.219 0.5-0.5zM16 21.5v-11c0-0.281-0.219-0.5-0.5-0.5h-1c-0.281 0-0.5 0.219-0.5 0.5v11c0 0.281 0.219 0.5 0.5 0.5h1c0.281 0 0.5-0.219 0.5-0.5zM7.5 6h7l-0.75-1.828c-0.047-0.063-0.187-0.156-0.266-0.172h-4.953c-0.094 0.016-0.219 0.109-0.266 0.172zM22 6.5v1c0 0.281-0.219 0.5-0.5 0.5h-1.5v14.812c0 1.719-1.125 3.187-2.5 3.187h-13c-1.375 0-2.5-1.406-2.5-3.125v-14.875h-1.5c-0.281 0-0.5-0.219-0.5-0.5v-1c0-0.281 0.219-0.5 0.5-0.5h4.828l1.094-2.609c0.313-0.766 1.25-1.391 2.078-1.391h5c0.828 0 1.766 0.625 2.078 1.391l1.094 2.609h4.828c0.281 0 0.5 0.219 0.5 0.5z"
    />
  </svg>
</div>
`;

exports[`<Icons /> should render <IconDelete /> with custom sizes 1`] = `
<div>
  <svg
    height="20"
    version="1.1"
    viewBox="0 0 22 28"
    width="20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M8 21.5v-11c0-0.281-0.219-0.5-0.5-0.5h-1c-0.281 0-0.5 0.219-0.5 0.5v11c0 0.281 0.219 0.5 0.5 0.5h1c0.281 0 0.5-0.219 0.5-0.5zM12 21.5v-11c0-0.281-0.219-0.5-0.5-0.5h-1c-0.281 0-0.5 0.219-0.5 0.5v11c0 0.281 0.219 0.5 0.5 0.5h1c0.281 0 0.5-0.219 0.5-0.5zM16 21.5v-11c0-0.281-0.219-0.5-0.5-0.5h-1c-0.281 0-0.5 0.219-0.5 0.5v11c0 0.281 0.219 0.5 0.5 0.5h1c0.281 0 0.5-0.219 0.5-0.5zM7.5 6h7l-0.75-1.828c-0.047-0.063-0.187-0.156-0.266-0.172h-4.953c-0.094 0.016-0.219 0.109-0.266 0.172zM22 6.5v1c0 0.281-0.219 0.5-0.5 0.5h-1.5v14.812c0 1.719-1.125 3.187-2.5 3.187h-13c-1.375 0-2.5-1.406-2.5-3.125v-14.875h-1.5c-0.281 0-0.5-0.219-0.5-0.5v-1c0-0.281 0.219-0.5 0.5-0.5h4.828l1.094-2.609c0.313-0.766 1.25-1.391 2.078-1.391h5c0.828 0 1.766 0.625 2.078 1.391l1.094 2.609h4.828c0.281 0 0.5 0.219 0.5 0.5z"
    />
  </svg>
</div>
`;

exports[`<Icons /> should render <IconPencil /> 1`] = `
<div>
  <svg
    height="16"
    version="1.1"
    viewBox="0 0 24 28"
    width="14"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M5.672 24l1.422-1.422-3.672-3.672-1.422 1.422v1.672h2v2h1.672zM13.844 9.5c0-0.203-0.141-0.344-0.344-0.344-0.094 0-0.187 0.031-0.266 0.109l-8.469 8.469c-0.078 0.078-0.109 0.172-0.109 0.266 0 0.203 0.141 0.344 0.344 0.344 0.094 0 0.187-0.031 0.266-0.109l8.469-8.469c0.078-0.078 0.109-0.172 0.109-0.266zM13 6.5l6.5 6.5-13 13h-6.5v-6.5zM23.672 8c0 0.531-0.219 1.047-0.578 1.406l-2.594 2.594-6.5-6.5 2.594-2.578c0.359-0.375 0.875-0.594 1.406-0.594s1.047 0.219 1.422 0.594l3.672 3.656c0.359 0.375 0.578 0.891 0.578 1.422z"
    />
  </svg>
</div>
`;

exports[`<Icons /> should render <IconPencil /> with custom sizes 1`] = `
<div>
  <svg
    height="20"
    version="1.1"
    viewBox="0 0 24 28"
    width="20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M5.672 24l1.422-1.422-3.672-3.672-1.422 1.422v1.672h2v2h1.672zM13.844 9.5c0-0.203-0.141-0.344-0.344-0.344-0.094 0-0.187 0.031-0.266 0.109l-8.469 8.469c-0.078 0.078-0.109 0.172-0.109 0.266 0 0.203 0.141 0.344 0.344 0.344 0.094 0 0.187-0.031 0.266-0.109l8.469-8.469c0.078-0.078 0.109-0.172 0.109-0.266zM13 6.5l6.5 6.5-13 13h-6.5v-6.5zM23.672 8c0 0.531-0.219 1.047-0.578 1.406l-2.594 2.594-6.5-6.5 2.594-2.578c0.359-0.375 0.875-0.594 1.406-0.594s1.047 0.219 1.422 0.594l3.672 3.656c0.359 0.375 0.578 0.891 0.578 1.422z"
    />
  </svg>
</div>
`;

exports[`<Icons /> should render <IconPlus /> 1`] = `
<div>
  <svg
    height="14"
    viewBox="0 0 24 24"
    width="14"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M18.984 12.984h-6v6h-1.97v-6h-6v-1.97h6v-6h1.97v6h6v1.97z"
    />
  </svg>
</div>
`;

exports[`<Icons /> should render <IconPlus /> with custom sizes 1`] = `
<div>
  <svg
    height="20"
    viewBox="0 0 24 24"
    width="20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M18.984 12.984h-6v6h-1.97v-6h-6v-1.97h6v-6h1.97v6h6v1.97z"
    />
  </svg>
</div>
`;

exports[`<Icons /> should render <IconUpload /> 1`] = `
<div>
  <svg
    height="16"
    viewBox="0 0 26 28"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M20 23c0-0.547-0.453-1-1-1s-1 0.453-1 1 0.453 1 1 1 1-0.453 1-1zM24 23c0-0.547-0.453-1-1-1s-1 0.453-1 1 0.453 1 1 1 1-0.453 1-1zM26 19.5v5c0 0.828-0.672 1.5-1.5 1.5h-23c-0.828 0-1.5-0.672-1.5-1.5v-5c0-0.828 0.672-1.5 1.5-1.5h6.672c0.422 1.156 1.531 2 2.828 2h4c1.297 0 2.406-0.844 2.828-2h6.672c0.828 0 1.5 0.672 1.5 1.5zM20.922 9.375c-0.156 0.375-0.516 0.625-0.922 0.625h-4v7c0 0.547-0.453 1-1 1h-4c-0.547 0-1-0.453-1-1v-7h-4c-0.406 0-0.766-0.25-0.922-0.625-0.156-0.359-0.078-0.797 0.219-1.078l7-7c0.187-0.203 0.453-0.297 0.703-0.297s0.516 0.094 0.703 0.297l7 7c0.297 0.281 0.375 0.719 0.219 1.078z"
    />
  </svg>
</div>
`;

exports[`<Icons /> should render <IconUpload /> with custom sizes 1`] = `
<div>
  <svg
    height="20"
    viewBox="0 0 26 28"
    width="20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M20 23c0-0.547-0.453-1-1-1s-1 0.453-1 1 0.453 1 1 1 1-0.453 1-1zM24 23c0-0.547-0.453-1-1-1s-1 0.453-1 1 0.453 1 1 1 1-0.453 1-1zM26 19.5v5c0 0.828-0.672 1.5-1.5 1.5h-23c-0.828 0-1.5-0.672-1.5-1.5v-5c0-0.828 0.672-1.5 1.5-1.5h6.672c0.422 1.156 1.531 2 2.828 2h4c1.297 0 2.406-0.844 2.828-2h6.672c0.828 0 1.5 0.672 1.5 1.5zM20.922 9.375c-0.156 0.375-0.516 0.625-0.922 0.625h-4v7c0 0.547-0.453 1-1 1h-4c-0.547 0-1-0.453-1-1v-7h-4c-0.406 0-0.766-0.25-0.922-0.625-0.156-0.359-0.078-0.797 0.219-1.078l7-7c0.187-0.203 0.453-0.297 0.703-0.297s0.516 0.094 0.703 0.297l7 7c0.297 0.281 0.375 0.719 0.219 1.078z"
    />
  </svg>
</div>
`;
